enclave-runtime/src/lib.rs

```diff
@@ -383,57 +385,164 @@ pub unsafe extern "C" fn get_state(
 	sgx_status_t::SGX_SUCCESS
 }
 
-/// Call this once at worker startup to initialize the TOP pool and direct invocation RPC server.
+/// Initialize generic enclave components.
 ///
-/// This function will run the RPC server on the same thread as it is called and will loop there.
-/// That means that this function will not return as long as the RPC server is running. The calling
-/// code should therefore spawn a new thread when calling this function.
+/// Call this once at startup. Needs to be the first component initializer to be called.
 #[no_mangle]
-pub unsafe extern "C" fn init_direct_invocation_server(
-	server_addr: *const u8,
-	server_addr_size: usize,
-) -> sgx_status_t {
-	let mut server_addr_encoded = slice::from_raw_parts(server_addr, server_addr_size);
+pub unsafe extern "C" fn init_enclave_base_components() -> sgx_status_t {
```

Comment:
Initialize the 'base' components. Components that are used by both the parentchain and sidechain. The STF executor is a 'base' component at the moment, because indirect invocation immediately executes the indirect calls. In the future, it will be a sidechain only component.